localstack/services/generic_proxy.py

```diff
@@ -186,6 +193,129 @@ def return_response(
+
+        pass
+
+    arn_regex = re.compile(
+        r"arn:"  # Prefix
+        r"(?P<Partition>(aws|aws-cn|aws-iso|aws-iso-b|aws-us-gov)*):"  # Partition
+        r"(?P<Service>[\w-]*):"  # Service (lambda, s3, ecs,...)
+        r"(?P<Region>[\w-]*):"  # Region (us-east-1, us-gov-west-1,...)
+        r"(?P<AccountID>[\w-]*):"  # AccountID
+        r"(?P<ResourcePath>"  # Combine the resource type and id to the ResourcePath
+        r"((?P<ResourceType>[\w-]*)[:/])?"  # ResourceType (optional, f.e. S3 bucket name)
+        r"(?P<ResourceID>[\w\-/*]*)"  # Resource ID (f.e. file name in S3)
+        r")"
+    )
+
+    def forward_request(
+        self, method: str, path: str, data: MessagePayload, headers: Headers
+    ) -> Optional[RoutingRequest]:
+        return RoutingRequest(
+            method=method,
+            path=self._adjust_partition_in_path(path),
+            data=self._adjust_partition(data),
+            headers=self._adjust_partition(headers),
+        )
+
+    def return_response(
+        self,
+        method: str,
+        path: str,
+        data: MessagePayload,
+        headers: Headers,
+        response: Response,
+    ) -> Optional[RoutingResponse]:
+        return RoutingResponse(
+            status_code=response.status_code,
+            content=self._adjust_partition(response.content),
+            headers=self._adjust_partition(response.headers),
+        )
+
+    def _adjust_partition_in_path(self, path: str):
+        """Adjusts the (still url encoded) URL path"""
+        parsed_url = urlparse(path)
+        decoded_query = parse_qs(parsed_url.query)
+        adjusted_path = self._adjust_partition(parsed_url.path)
+        adjusted_query = self._adjust_partition(decoded_query)
+        encoded_query = urlencode(adjusted_query, doseq=True)
+        return f"{adjusted_path}{('?' + encoded_query) if encoded_query else ''}"
+
+    def _adjust_partition(self, source):
+        # Call this function recursively if we get a dictionary or a list
+        if isinstance(source, dict):
+            result = {}
+            for k, v in source.items():
+                result[k] = self._adjust_partition(v)
+            return result
+        if isinstance(source, list):
+            result = []
+            for v in source:
+                result.append(self._adjust_partition(v))
+            return result
+        elif isinstance(source, bytes):
+            decoded = to_str(source)
+            adjusted = self._adjust_partition(decoded)
+            return to_bytes(adjusted)
+        elif not isinstance(source, str):
+            # Ignore any other types
+            return source
+        return self.arn_regex.sub(PartitionAdjustingProxyListener._adjust_match, source)
+
+    @staticmethod
+    def _adjust_match(match: Match):
+        region = match.group("Region")
+        try:
+            partition = PartitionAdjustingProxyListener._get_partition_for_region(region)
```

Comment:
like daniel noted, the forwarded partition should always be aws for incoming requests. 

only outgoing need this region => partition mapping

Reply:
Done.